plugins/lttng-utils/debug-info.c

```diff
@@ -726,22 +913,933 @@ void debug_info_handle_event(FILE *err, const bt_event *event,
+		if (full_path) {
+			g_string_printf(curr_field_str, "%s:%" PRId64,
+					dbg_info_src->src_path,
+					dbg_info_src->line_no);
+		} else {
+			g_string_printf(curr_field_str, "%s:%" PRId64,
+					dbg_info_src->short_src_path,
+					dbg_info_src->line_no);
+		}
+	}
+}
+
+static
+void fill_debug_info_field(struct debug_info *debug_info, int64_t vpid,
+		uint64_t ip, bt_field *debug_info_field)
+{
+	struct debug_info_source * dbg_info_src;
+	const bt_field_class *debug_info_fc;
+	bt_field_status status;
+	uint64_t i, nb_member;
+	GString *curr_field_str;
+
+	BT_ASSERT(bt_field_get_class_type(debug_info_field) ==
+			BT_FIELD_CLASS_TYPE_STRUCTURE);
+
+	debug_info_fc = bt_field_borrow_class_const(debug_info_field);
+
+	nb_member = bt_field_class_structure_get_member_count(debug_info_fc);
+	BT_ASSERT(nb_member == 3);
+
+	dbg_info_src = debug_info_query(debug_info, vpid, ip);
+
+	/* Start with an empty string. */
+	curr_field_str = g_string_new(NULL);
+	for (i = 0; i < nb_member; i++) {
+		const char *curr_field_name;
+		const bt_field_class *curr_fc;
+		bt_field *curr_field;
+
+		bt_field_class_structure_borrow_member_by_index_const(
+				debug_info_fc, i, &curr_field_name, &curr_fc);
+		curr_field = bt_field_structure_borrow_member_field_by_index(
+					debug_info_field, i);
+
+		BT_ASSERT(bt_field_get_class_type(curr_field) ==
+				BT_FIELD_CLASS_TYPE_STRING);
+		/*
+		 * If no debug information is available, simply keep the field
+		 * empty.
+		 */
+		if (dbg_info_src) {
+			if (!strcmp(curr_field_name, "bin")) {
+				fill_debug_info_bin_field(dbg_info_src,
+						debug_info->comp->arg_full_path,
+						curr_field_str);
+			} else if (!strcmp(curr_field_name, "func")) {
+				fill_debug_info_func_field(dbg_info_src,
+						curr_field_str);
+			} else if (!strcmp(curr_field_name, "src")) {
+				fill_debug_info_src_field(dbg_info_src,
+						debug_info->comp->arg_full_path,
+						curr_field_str);
+			}
+		}
+
+		/* Set field to the string crafted from the debug information. */
+		status = bt_field_string_set_value(curr_field, curr_field_str->str);
+		if (status != BT_FIELD_STATUS_OK) {
+			BT_LOGE("Cannot set \"%s\" curr_field_str field's value: "
+				"str-fc-addr=%p, str-value=\"%s\"", curr_field_name,
+				curr_field, curr_field_str->str);
+		}
+		g_string_truncate(curr_field_str, 0);
+	}
+	g_string_free(curr_field_str, TRUE);
+}
+
+static inline
+void fill_debug_info_event(struct debug_info_iterator *debug_it,
+		const bt_event *in_event, bt_event *out_event)
+{
+	bt_field *out_common_ctx_field, *out_debug_info_field;
+	struct debug_info *debug_info;
+	char *debug_info_field_name;
+
+	debug_info_field_name =
+		debug_it->debug_info_component->arg_debug_info_field_name;
+
+	/* Borrow the debug-info field. */
+	out_common_ctx_field = bt_event_borrow_common_context_field(out_event);
+	out_debug_info_field = bt_field_structure_borrow_member_field_by_name(
+				out_common_ctx_field, debug_info_field_name);
+
+	if (out_debug_info_field) {
```

Comment:
What should we do the debug-info field is already present? should we append to it? create a debug-info-2 field ? 
here is the metadata part of the previous implementation: https://github.com/eepp/babeltrace/blob/optimalize/plugins/lttng-utils/copy.c#L829

Here is my suggestion:

- If there is a `debug-info` struct field class and any of its regular members (`bin`, `func`, `src`) are present, debug-info should act as a passthrough (don't add field classes to the common context) so not to override the work of a upstream component.

- If there is a `debug-info` struct field class and none of its regular members (`bin`, `func` or `src`) are present , debug-info should add the new field classes (`bin`, `func`, `src`) to the `debug-info` struct.

Reply:
I would be more strict than that: if there's a `debug-info` member (actually that's just the default name of the `debug_info_field_name` string), whatever its field class, disable the filter for this whole stream class (copy, but don't do any debug information lookup or modification to the copied stream).

The user always has the choice to name this field another way should there be a name clash.

Reply:
Make sense.
So act as a passthrough if `debug-info` field class already present in the event common context.

Reply:
Yes, a passthrough, but still copying because there could be another stream class in the same trace class (now or in the future) with a compatible common event context field class.

Reply:
yes. that's what i meant to say.